assets/js/boorujs.ts

```diff
@@ -0,0 +1,214 @@
+  },
+  preventdefault() {
+    /* The existence of this entry is enough */
+  },
+
+  copy(data) {
+    const target = $<HTMLInputElement | HTMLTextAreaElement>(data.value);
+    if (target) {
+      target.select();
+      document.execCommand('copy');
+    }
+  },
+
+  inputvalue(data) {
+    const target = $<HTMLInputElement | HTMLTextAreaElement>(data.value);
+    const setValue = data.el.dataset.setValue;
+    if (target && setValue !== undefined) {
+      target.value = setValue;
+    }
+  },
+
+  selectvalue(data) {
+    const target = $<HTMLSelectElement | HTMLInputElement>(data.value);
+    const checked = $<HTMLInputElement>(':checked', data.el);
+    const setValue = checked?.dataset.setValue;
+
+    if (target && setValue !== undefined) {
+      target.value = setValue;
+    }
+  },
+
+  checkall(data) {
+    $$<HTMLInputElement>(`${data.value} input[type=checkbox]`).forEach(c => {
+      c.checked = !c.checked;
+    });
+  },
+
+  addtag(data) {
+    const targetContainer = data.el.closest<HTMLElement>('[data-target]');
+    const tagName = data.el.dataset.tagName;
+
+    if (targetContainer && tagName && targetContainer.dataset.target) {
+      const target = $<HTMLInputElement | HTMLTextAreaElement>(targetContainer.dataset.target);
+
+      if (target) {
+        addTag(target, tagName);
+      }
+    }
+  },
+
+  hideParent(data) {
+    const base = data.el.closest<HTMLElement>(data.value);
+    if (base) {
+      base.classList.add('hidden');
+    }
+  },
+
+  tab(data) {
+    const block = data.el.parentNode?.parentNode;
+    if (!(block instanceof HTMLElement)) return;
+
+    const newTab = $<HTMLElement>(`.block__tab[data-tab="${data.value}"]`, block);
+    const loadTab = data.el.dataset.loadTab;
+
+    // Switch tab
+    const selectedTab = $<HTMLElement>('.selected', block);
+    if (selectedTab) {
+      selectedTab.classList.remove('selected');
+    }
+    data.el.classList.add('selected');
+
+    // Switch contents
+    actions.tabHide({ ...data, base: block, value: '.block__tab' });
+    actions.show({ ...data, base: block, value: `.block__tab[data-tab="${data.value}"]` });
+
+    // If the tab has a 'data-load-tab' attribute, load and insert the content
+    if (loadTab && newTab && !newTab.dataset.loaded) {
+      fetchHtml(loadTab)
+        .then(handleError)
+        .then(response => response.text())
+        .then(response => (newTab.innerHTML = response))
+        .then(() => (newTab.dataset.loaded = 'true'))
+        .catch(() => (newTab.textContent = 'Error!'));
+    }
+  },
+};
+
+// Use this function to apply a callback to elements matching the selectors
+function selectorCb(base: ParentNode = document, selector: string, cb: (el: Element) => void) {
+  $$(selector, base).forEach(cb);
+}
+
+function selectorCbChildren(base: ParentNode = document, selector: string, cb: (el: Element) => void) {
+  const sel = $$(selector, base);
+
+  for (const el of Array.from(base.children)) {
```

Comment:
Array.from shouldn't be needed I think, it should just let you iterate the children list